phpunit/class-reusable-blocks-render-test.php

```diff
@@ -0,0 +1,121 @@
+<?php
+/**
+ * Reusuable block rendering tests.
+ *
+ * @package Gutenberg
+ */
+
+/**
+ * Tests reusuable block rendering.
+ */
+class Reusuable_Blocks_Render_Test extends WP_UnitTestCase {
+	/**
+	 * Fake user ID.
+	 *
+	 * @var int
+	 */
+	protected static $user_id;
+
+	/**
+	 * Fake block ID.
+	 *
+	 * @var int
+	 */
+	protected static $block_id;
+
+	/**
+	 * Fake post ID.
+	 *
+	 * @var int
+	 */
+	protected static $post_id;
+
+	/**
+	 * Create fake data before tests run.
+	 *
+	 * @param WP_UnitTest_Factory $factory Helper that creates fake data.
+	 */
+	public static function wpSetUpBeforeClass( $factory ) {
+		self::$user_id = $factory->user->create( array(
+			'role' => 'editor',
+		) );
+
+		self::$post_id = $factory->post->create( array(
+			'post_author'  => self::$user_id,
+			'post_type'    => 'post',
+			'post_status'  => 'publish',
+			'post_title'   => 'Test Post',
+			'post_content' => '<p>Hello world!</p>',
+		) );
+
+		self::$block_id = $factory->post->create( array(
+			'post_author'  => self::$user_id,
+			'post_type'    => 'wp_block',
+			'post_status'  => 'publish',
+			'post_title'   => 'Test Block',
+			'post_content' => '<!-- wp:core/paragraph --><p>Hello world!</p><!-- /wp:core/paragraph -->',
+		) );
+	}
+
+	/**
+	 * Delete fake data after tests run.
+	 */
+	public static function wpTearDownAfterClass() {
+		wp_delete_post( self::$block_id, true );
+		wp_delete_post( self::$post_id, true );
+		self::delete_user( self::$user_id );
+	}
+
+	/**
+	 * Set up.
+	 */
+	public function setUp() {
+		parent::setUp();
+
+		$this->registry = new WP_Block_Type_Registry();
+		$main_registry  = WP_Block_Type_Registry::get_instance();
+
+		$this->registry->register( 'core/block', $main_registry->get_registered( 'core/block' ) );
```

Comment:
We can avoid having this proxy registry by using the main registry in the tests.

That is, we could delete this `setUp()` entirely and change our tests to look like this:

```php
public function test_render() {
    $block_type = WP_Block_Type_Registry::get_instance()->get_registered( 'core/block' );
    $output = $block_type->render( array( 'ref' => self::$block_id ) );
    $this->assertSame( '<p>Hello world!</p>', $output );
}
```